### PR TITLE
SF-1335: Ensure sessionId is only present in search requests

### PR DESCRIPTION
**Pull request by JoeyDeol**

### CODE REVIEW

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts [hunk 1]
```diff
@@ -55,21 +59,22 @@ suite('requests helpers', ({ expect, stub, spy }) => {
...
+        sessionId,
         skip: originalSkip
       });
     });
 
     it('should decrease page size to prevent exceeding MAX_RECORDS', () => {
       stub(Selectors, 'config').returns({ search: {} });
 
-      const { pageSize, skip } = RequestHelpers.search(<any>{});
+      const { pageSize, skip } = RequestHelpers.search(state);
```
Comment (jrmykolyn) [inline packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: This doesn't appear to be related to the update. Do we need to adjust this line?

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts [hunk 2]
```diff
@@ -80,7 +85,7 @@ suite('requests helpers', ({ expect, stub, spy }) => {
       const extractLanguage = stub(ConfigAdapter, 'extractLanguage').returns(language);
       stub(Selectors, 'config').returns({ search: {} });
 
-      const request = RequestHelpers.search(<any>{});
+      const request = RequestHelpers.search(state);
```
Comment (jrmykolyn) [inline packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: Same as above.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts [hunk 3]
```diff
@@ -91,14 +96,13 @@ suite('requests helpers', ({ expect, stub, spy }) => {
       requestSortAdapter.returns(sort);
       stub(Selectors, 'config').returns({ search: {} });
 
-      const request = RequestHelpers.search(<any>{});
+      const request = RequestHelpers.search(state);
```
Comment (jrmykolyn) [inline packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: Same as above.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts [hunk 4]
```diff
@@ -35,11 +35,15 @@ suite('requests helpers', ({ expect, stub, spy }) => {
...
     const refinements = [{ a: 'b' }, { c: 'd' }];
+    let state;
+    let sessionId;
     let sortSelector: sinon.SinonStub;
     let requestSortAdapter: sinon.SinonStub;
     let pastPurchaseBiasingAdapter: sinon.SinonStub;
 
     beforeEach(() => {
+      state = <any>{ session: { sessionId: 'foo' } };
+      sessionId = <any>RequestHelpers.extractSessionId(state);
```
[the remark this answers is not in the document]
Comment (jmbtutor) [reply packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: You may also want to consider stubbing `extractSessionId` instead of assuming the presence of `sessionId` in state, although you could argue that `extractSessionId` is a helper function and should be okay not being stubbed.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts [hunk 5]
```diff
@@ -116,7 +120,7 @@ suite('requests helpers', ({ expect, stub, spy }) => {
       const overrideRequest = { pageSize, skip };
       stub(Selectors, 'config').returns({});
 
-      const request = RequestHelpers.search(<any>{}, overrideRequest);
+      const request = RequestHelpers.search(state, overrideRequest);
```
Comment (jrmykolyn) [inline packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: Same as above.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts [hunk 6]
```diff
@@ -581,17 +585,14 @@ suite('requests helpers', ({ expect, stub, spy }) => {
...
-    it('should return a function which applies the sessionId to the request', () => {
-      const req = { a: 'b' };
+  describe('extractSessionId()', () => {
+    it('should return sessionId', () => {
       const sessionId = 'foo';
       const state: any = { c: 'd', session: { sessionId } };
 
-      const attachSessionId = RequestHelpers.attachSessionId(state);
-      const requestWithSessionId = attachSessionId(req);
+      const extractedSessionId = RequestHelpers.extractSessionId(state);
```
Comment (jrmykolyn) [inline packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: I would just inline this.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts (hunk 4 above)
Comment (jrmykolyn) [inline packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: I don't think we want to derive `sessionId` *from* `extractSessionid()`. Rather, we should be doing defining `sessionId` as some primitive value, then defining `state` using the resulting variable.

File: packages/@storefront/flux-capacitor/CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased] [patch]
+### Changed
+- SF-1335: `sessionId` will now only be included in `search` request objects.
```
Comment (jmbtutor) [inline packages/@storefront/flux-capacitor/CHANGELOG.md]: I would say something like:

```
`sessionId` will now only be sent with search requests.
```

Comment (JoeyDeol) [reply packages/@storefront/flux-capacitor/CHANGELOG.md]: Updated, resolving this conversation.

File: packages/@storefront/flux-capacitor/CHANGELOG.md [hunk 8]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
...
+  - `attachSessionId()` has been changed to `extractSessionId()`.
+  - The `BuildFunction()` in the case of `search` now uses `extractSessionId()` to retrieve the `sessionId` from the app store.
```
[the remark this answers is not in the document]
Comment (JoeyDeol) [reply packages/@storefront/flux-capacitor/CHANGELOG.md]: Will do. Resolving conversation.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts (hunk 6 above)
Comment (JoeyDeol) [reply packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: Will do. Resolving.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (JoeyDeol) [reply packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: I see, I will make these changes. Resolving conversation.

Comment (JoeyDeol) [reply packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: Right, I will make the changes. Resolving conversation.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts (hunk 1 above)
Comment (JoeyDeol) [reply packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: You're right it's not related, was required before my update to the way I was updating the test suite. Reverting these changes.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts (hunk 2 above)
Comment (JoeyDeol) [reply packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: Reverting this change.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts (hunk 3 above)
Comment (JoeyDeol) [reply packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: Reverting this change.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts (hunk 5 above)
Comment (JoeyDeol) [reply packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: Reverting this change.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts [hunk 9]
```diff
@@ -50,26 +54,28 @@ suite('requests helpers', ({ expect, stub, spy }) => {
...
+        sessionId,
         skip: originalSkip
       });
     });
 
     it('should decrease page size to prevent exceeding MAX_RECORDS', () => {
       stub(Selectors, 'config').returns({ search: {} });
 
-      const { pageSize, skip } = RequestHelpers.search(<any>{});
+      const { pageSize, skip } = RequestHelpers.search(<any> {});
```
Comment (jrmykolyn) [inline packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: It pains me to say this, but it looks like our convention is no space between the cast type and the cart target (ie. `<any>{}`.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts [hunk 10]
```diff
@@ -80,7 +86,7 @@ suite('requests helpers', ({ expect, stub, spy }) => {
       const extractLanguage = stub(ConfigAdapter, 'extractLanguage').returns(language);
       stub(Selectors, 'config').returns({ search: {} });
 
-      const request = RequestHelpers.search(<any>{});
+      const request = RequestHelpers.search(<any> {});
```
Comment (jrmykolyn) [inline packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: Same as above... and below.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts (hunk 9 above)
Comment (JoeyDeol) [reply packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: My bad. Updated to remove the space.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts (hunk 10 above)
Comment (JoeyDeol) [reply packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: Updated.

File: packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts [hunk 11]
```diff
@@ -50,18 +54,20 @@ suite('requests helpers', ({ expect, stub, spy }) => {
...
       stub(Selectors, 'page').returns(page);
       stub(Selectors, 'pageSize').returns(originalPageSize);
+      stub(RequestHelpers, 'extractSessionId').returns(sessionId);
     });
 
     it('should build out request', () => {
       stub(Selectors, 'config').returns({});
 
-      expect(RequestHelpers.search(<any>{})).to.eql({
+      expect(RequestHelpers.search(state)).to.eql({
```
Comment (jmbtutor) [inline packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: This doesn't need to change.

Comment (JoeyDeol) [reply packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts]: Removed, resolving this conversation.
